modules/table/src/test/java/org/apache/ignite/internal/table/distributed/raft/PartitionCommandListenerTest.java

```diff
@@ -206,6 +215,42 @@ public void testInsertRowsBatchedAndCheck() {
         readAndCheck(false);
     }
 
+    /**
+     * The test checks that {@link PartitionListener#onSnapshotSave(Path, Consumer)} propagates
+     * the maximal last applied index among storages to all storages.
+     */
+    @Test
+    public void testOnSnapshotSavePropagateLastAppliedIndex() {
+        ReplicaService replicaService = mock(ReplicaService.class, RETURNS_DEEP_STUBS);
+
+        TestConcurrentHashMapTxStateStorage txStateStorage = new TestConcurrentHashMapTxStateStorage();
+
+        TestPartitionDataStorage partitionDataStorage = new TestPartitionDataStorage(mvPartitionStorage);
+
+        PartitionListener testCommandListener = new PartitionListener(
+                partitionDataStorage,
+                txStateStorage,
+                new TxManagerImpl(replicaService, new HeapLockManager(), new HybridClock()),
+                () -> Map.of(pkStorage.id(), pkStorage)
+        );
+
+        txStateStorage.lastAppliedIndex(3L);
+
+        partitionDataStorage.lastAppliedIndex(5L);
+
+        AtomicLong counter = new AtomicLong(0);
+
+        testCommandListener.onSnapshotSave(workDir, (throwable) -> {
+            counter.incrementAndGet();
+        });
+
+        assertEquals(1L, counter.get());
+
+        assertEquals(5L, partitionDataStorage.lastAppliedIndex());
```

Comment:
Here it is not checked what exactly the maximum was taken, but only for `partitionDataStorage` let's run **onSnapshotSave** one more time but with a maximum for `txStateStorage`.

Reply:
fix it